apps/api/src/app/blueprint/e2e/get-grouped-blueprints.e2e.ts

```diff
@@ -0,0 +1,43 @@
+import { expect } from 'chai';
+
+import { UserSession } from '@novu/testing';
+import { NotificationTemplateRepository, EnvironmentRepository } from '@novu/dal';
+import { createTemplateFromBlueprint } from './create-blueprint.e2e';
+import { GroupedBlueprintResponse } from '../dto/grouped-blueprint.response.dto';
+
+describe('Get grouped notification template blueprints - /blueprints/group-by-category (GET)', async () => {
+  let session: UserSession;
+  const notificationTemplateRepository: NotificationTemplateRepository = new NotificationTemplateRepository();
+  const environmentRepository: EnvironmentRepository = new EnvironmentRepository();
+
+  before(async () => {
+    session = new UserSession();
+    await session.initialize();
+  });
+
+  it('should get the grouped blueprints', async function () {
+    const prodEnv = await getProductionEnvironment();
+
+    await createTemplateFromBlueprint({ session, notificationTemplateRepository, prodEnv });
+
+    const data = await session.testAgent.get(`/v1/blueprints/group-by-category`).send();
+
+    expect(data.statusCode).to.equal(200);
+
+    const groupedBlueprints = data.body.data as GroupedBlueprintResponse[];
+
+    expect(groupedBlueprints[0].name).to.equal('General');
+
+    for (const grouped of groupedBlueprints) {
+      for (const blueprint of grouped.blueprints) {
```

Comment:
```suggestion
    for (const group of groupedBlueprints) {
      for (const blueprint of group.blueprints) {
```

Reply:
I feel like we should also check some base fields that are needed for the FE, like id, name, description at least

Reply:
Good suggestion, I will add some.